File: 0860-lemonade-change/0860-lemonade-change.cpp
class Solution {
public:
    bool lemonadeChange(vector<int>& bills) {
        int collectedMoney=0;
        int dollar5=0;
        int dollar10=0;
        for(int i=0;i<bills.size();i++) {
           if(bills[i]==10){
               if(dollar5>0) {
                   dollar5--;
                   dollar10++;
               }else {
                   return false;
               }
           } else if(bills[i]==20) {
               if(dollar10>0 && dollar5>0) {
                   dollar10--;
                   dollar5--;
               } else if(dollar5>=3) {
                   dollar5-=3;
               } else {
                   return false;
               }
               
           } else {
               dollar5++;
           }
            
        }
        return true;
        
    }
};

File: 0860-lemonade-change/0860-lemonade-change.java
class Solution {
    public boolean lemonadeChange(int[] bills) {
        int fiveBillCollected = 0;
        int tenBillCollected =0;
        for(int index =0 ;index < bills.length ;index ++){
            if(bills[index]==5){
                fiveBillCollected++;
            }else if(bills [index] ==10){
                if(fiveBillCollected > 0){
                    tenBillCollected ++;
                    fiveBillCollected --;
                }else{
                    System.out.println(index +" " +"A");
                    return false;
                }
                
                
            }else{
                if(fiveBillCollected > 0 && tenBillCollected >0){
                    tenBillCollected --;
                    fiveBillCollected --;
                }else if(fiveBillCollected > 2){
                    fiveBillCollected -=3;
                }else{
                    System.out.println(index +" " +"B");
                    return false;
                }
            }
        }
        return true;
        
    }
}